pkg/querier/tripperware/priority.go

```diff
@@ -2,65 +2,29 @@ package tripperware
 
 import (
 	"errors"
-	"net/http"
-	"strings"
 	"time"
 
-	"github.com/prometheus/prometheus/promql/parser"
-
-	"github.com/cortexproject/cortex/pkg/util"
 	"github.com/cortexproject/cortex/pkg/util/validation"
 )
 
 var (
-	errParseExpr = errors.New("failed to parse expr")
+	errQueryPriorityDisabled = errors.New("query priority disabled")
+	errEmptyQueryString      = errors.New("empty query string")
 )
 
-func GetPriority(r *http.Request, userID string, limits Limits, now time.Time, lookbackDelta time.Duration) (int64, error) {
-	isQuery := strings.HasSuffix(r.URL.Path, "/query")
-	isQueryRange := strings.HasSuffix(r.URL.Path, "/query_range")
-	queryPriority := limits.QueryPriority(userID)
-	query := r.FormValue("query")
-
-	if (!isQuery && !isQueryRange) || !queryPriority.Enabled || query == "" {
-		return 0, nil
+func GetPriority(query string, minTime, maxTime int64, now time.Time, queryPriority validation.QueryPriority) (int64, error) {
```

Comment:
Do we still need to return error in the response? To me both cases `!queryPriority.Enabled` and `query == ""` are not really error. 
We can just use the default priority or 0. WDYT?

Because we have log at https://github.com/cortexproject/cortex/pull/5758/files#diff-8408ad88dab157cda0e6650bdae6fc2820d14af9b3e31df4798a86e5e300b26eR175, and considering the two cases it might log too much

Reply:
This is more of a preventative measure, just in case other functions in the future call `GetPriority` before checking the config or query string itself. Currently, we do have checks prior to calling `GetPriority`, such that the checks are redundant + error will never be thrown. It's also debug level, so I still lean towards keeping it.

Reply:
It doesn't make much sense that we return error when queryPriority is not enabled. If we don't enable priority, just return the default value and it is a normal case rather than an error. Same for empty query string since it will be a noop at querier basically.

> Currently, we do have checks prior to calling GetPriority, such that the checks are redundant + error will never be thrown

I feel those are normal cases and not really an error, so I think we can remove it and no need to reserve it for future.

Reply:
ah ok, will make the change